### DISCUSSION TITLE
How to commit when job only run for tags?

### DISCUSSION THREAD

**Discussion by dhruvinsh** [caksoylar/keymap-drawer#128] [Q&A]
Hi,
I run the keymap-drawer ci on if tag has been pushed. 

in such case i can't amend the commit message, since it tag version not a branch.

I tried with commit_message but it fail as well, 
```
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>
```

Is there any way to commit to specific branch?

**Comment by caksoylar** [comment]
Apparently there is an option to specify the branch name to push to on the commit action: https://github.com/stefanzweifel/git-auto-commit-action/blob/master/action.yml#L11
We can add such an input to the workflow to pass through to the commit action.

I feel like this will have issues when you run it on tag push, though. If you add a new commit, it will not be the commit that the tag is pointing to. If you amend the commit, it will similarly not match the exact SHA that the tag is assigned to. Maybe you'd want to use `destination: artifact` for this? https://github.com/caksoylar/keymap-drawer/blob/main/.github/workflows/draw-zmk.yml#L62


If you use the tag to create a release, maybe there is a way to add the images to the release files instead.

**Reply by dhruvinsh** [reply]
> Maybe you'd want to use destination: artifact for this?

Yes I looked at it but with the readme files fetching the local assets would be broken unless I push files manually. 

> Apparently there is an option to specify the branch name to push to

Yes the upstream supports such option and that would be great that we can expose it as well but,

> I feel like this will have issues when you run it on tag push, though. 

Good point. Basically i can just use file change trigger to run draw job and skip it when tags are pushed. That would be the best option.

**Reply by caksoylar** [reply]
> Good point. Basically i can just use file change trigger to run draw job and skip it when tags are pushed. That would be the best option.

That sounds reasonable to me if you want the output files to live in the repo, yes.
